src/main/kotlin/com/demonwav/mcdev/platform/mixin/action/GenerateAccessorHandler.kt

```diff
@@ -0,0 +1,496 @@
+                countAccessorMixins(project, it.qualifiedClient) +
+                countAccessorMixins(project, it.qualifiedServer)
+        }
+
+        if (config == null) {
+            // TODO: generate the mixin configuration file (modding platform dependent)
+            val message = "There is no matching mixin configuration file found in the project. " +
+                "Please create one and try again."
+            Messages.showInfoMessage(project, message, "Generate Accessor/Invoker")
+            return null
+        }
+
+        val defaultPkg = config.pkg ?: ""
+        val defaultName = "${targetClass.name}Accessor"
+        val defaultModule = config.file?.let { ModuleUtil.findModuleForFile(it, project) }
+
+        val dialog = CreateClassDialog(
+            project,
+            "Create Accessor Mixin",
+            defaultName,
+            defaultPkg,
+            CreateClassKind.CLASS,
+            true,
+            defaultModule
+        )
+        if (!dialog.showAndGet()) {
+            return null
+        }
+        val pkg = dialog.targetDirectory ?: return null
+        val name = dialog.className
+
+        var clazz: PsiClass? = null
+        WriteCommandAction.writeCommandAction(project)
+            .withName("Generate Accessor/Invoker")
+            .withGroupId("Generate Accessor/Invoker")
+            .run<RuntimeException> {
+                IdeDocumentHistory.getInstance(project).includeCurrentPlaceAsChangePlace()
+                try {
+                    clazz = JavaDirectoryService.getInstance().createInterface(pkg, name)
+                } catch (e: IncorrectOperationException) {
+                    invokeLater {
+                        val message = "${CodeInsightBundle.message(
+                            "intention.error.cannot.create.class.message",
+                            name
+                        )}\n${e.localizedMessage}"
+                        Messages.showErrorDialog(
+                            project,
+                            message,
+                            CodeInsightBundle.message("intention.error.cannot.create.class.title")
+                        )
+                    }
+                    return@run
+                }
+                val factory = JavaPsiFacade.getElementFactory(project)
+                val targetAccessible = targetClass.modifierList?.hasExplicitModifier(PsiModifier.PUBLIC) == true &&
+                    targetClass.name != null
+                val annotationText = if (targetAccessible) {
+                    "@${MixinConstants.Annotations.MIXIN}(${targetClass.qualifiedName}.class)"
+                } else {
+                    "@${MixinConstants.Annotations.MIXIN}(targets=\"${targetClass.fullQualifiedName}\")"
+                }
+                val annotation = factory.createAnnotationFromText(annotationText, clazz)
+                AddAnnotationFix(MixinConstants.Annotations.MIXIN, clazz!!, annotation.parameterList.attributes)
```

Comment:
It's not possible for clazz to be null at this point. It's assigned from `JavaDirectoryService.createInstance` in the try block, which is `@NotNull`. And if the exception is caught, then it returns out of the block.

Reply:
I generally like avoiding `!!` in general when possible, but since you've correctly stated it's not possible to be `null`, I try to find another way to re-arrange the code to show that. I took another stab at that and this is what I came up with.

```kotlin
return WriteCommandAction.writeCommandAction(project)
    .withName("Generate Accessor/Invoker")
    .withGroupId("Generate Accessor/Invoker")
    .compute<PsiClass, RuntimeException> {
        IdeDocumentHistory.getInstance(project).includeCurrentPlaceAsChangePlace()
        val clazz = try {
            JavaDirectoryService.getInstance().createInterface(pkg, name)
        } catch (e: IncorrectOperationException) {
            invokeLater {
                val message = "${CodeInsightBundle.message(
                    "intention.error.cannot.create.class.message",
                    name
                )}\n${e.localizedMessage}"
                Messages.showErrorDialog(
                    project,
                    message,
                    CodeInsightBundle.message("intention.error.cannot.create.class.title")
                )
            }
            return@compute null
        }
        val factory = JavaPsiFacade.getElementFactory(project)
        val targetAccessible = targetClass.modifierList?.hasExplicitModifier(PsiModifier.PUBLIC) == true &&
            targetClass.name != null
        val annotationText = if (targetAccessible) {
            "@${MixinConstants.Annotations.MIXIN}(${targetClass.qualifiedName}.class)"
        } else {
            "@${MixinConstants.Annotations.MIXIN}(targets=\"${targetClass.fullQualifiedName}\")"
        }
        val annotation = factory.createAnnotationFromText(annotationText, clazz)
        AddAnnotationFix(MixinConstants.Annotations.MIXIN, clazz, annotation.parameterList.attributes)
            .applyFix()

        val module = clazz.findModule() ?: return@compute null
        val configToWrite = MixinModule.getBestWritableConfigForMixinClass(
            project,
            GlobalSearchScope.moduleScope(module),
            clazz.fullQualifiedName ?: ""
        )
        configToWrite?.qualifiedMixins?.add(clazz.fullQualifiedName)

        return@compute clazz
    }
```

We can use the `computer` method in combination with using the `try { ... } catch { ... }` block as an expression and I think that's a cleaner solution.

Reply:
I generally like avoiding `!!` in general when possible, but since you've correctly stated it's not possible to be `null`, I try to find another way to re-arrange the code to show that. I took another stab at that and this is what I came up with.

```kotlin
return WriteCommandAction.writeCommandAction(project)
    .withName("Generate Accessor/Invoker")
    .withGroupId("Generate Accessor/Invoker")
    .compute<PsiClass, RuntimeException> {
        IdeDocumentHistory.getInstance(project).includeCurrentPlaceAsChangePlace()
        val clazz = try {
            JavaDirectoryService.getInstance().createInterface(pkg, name)
        } catch (e: IncorrectOperationException) {
            invokeLater {
                val message = "${CodeInsightBundle.message(
                    "intention.error.cannot.create.class.message",
                    name
                )}\n${e.localizedMessage}"
                Messages.showErrorDialog(
                    project,
                    message,
                    CodeInsightBundle.message("intention.error.cannot.create.class.title")
                )
            }
            return@compute null
        }
        val factory = JavaPsiFacade.getElementFactory(project)
        val targetAccessible = targetClass.modifierList?.hasExplicitModifier(PsiModifier.PUBLIC) == true &&
            targetClass.name != null
        val annotationText = if (targetAccessible) {
            "@${MixinConstants.Annotations.MIXIN}(${targetClass.qualifiedName}.class)"
        } else {
            "@${MixinConstants.Annotations.MIXIN}(targets=\"${targetClass.fullQualifiedName}\")"
        }
        val annotation = factory.createAnnotationFromText(annotationText, clazz)
        AddAnnotationFix(MixinConstants.Annotations.MIXIN, clazz, annotation.parameterList.attributes)
            .applyFix()

        val module = clazz.findModule() ?: return@compute null
        val configToWrite = MixinModule.getBestWritableConfigForMixinClass(
            project,
            GlobalSearchScope.moduleScope(module),
            clazz.fullQualifiedName ?: ""
        )
        configToWrite?.qualifiedMixins?.add(clazz.fullQualifiedName)

        return@compute clazz
    }
```

We can use the `computer` method in combination with using the `try { ... } catch { ... }` block as an expression and I think that's a cleaner solution.